pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -258,3 +266,21 @@ func (dv *openShiftClusterDynamicValidator) validateSubnet(ctx context.Context,
 
 	return net, nil
 }
+
+// validateVnet checks that the vnet does not have custom dns servers set
+func (dv *openShiftClusterDynamicValidator) validateVnet(ctx context.Context, m utilvnet.Manager, oc *api.OpenShiftCluster) error {
+	vnetID, _, err := subnet.Split(oc.Properties.MasterProfile.SubnetID)
+	if err != nil {
+		return err
+	}
+	vnet, err := m.Get(ctx, vnetID)
+	if err != nil {
+		return err
+	}
+	servers := to.StringSlice(vnet.DhcpOptions.DNSServers)
+	if len(servers) > 0 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "properties.masterProfile.subnetId", "The provided vnetID  '%s' is invalid: custom DNS servers are not supported.", vnetID)
```

Comment:
`return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "", "The provided vnet '%s' is invalid: custom DNS servers are not supported.", vnetID)`